### ISSUE TITLE
Multi-words terms with quotation sign(") not handled by thesaurus

### ISSUE THREAD

**Issue by rohitkundale** [Smile-SA/elasticsuite#1728]
If a multi-word synonym or expansion rule contains Inch sign(") multiple times, indexing the thesaurus fails with the error
### Preconditions

`{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"failed to build synonyms"}],"type":"illegal_argument_exception","reason":"failed to build synonyms","caused_by":{"type":"parse_exception","reason":"Invalid synonym rule at line 38","caused_by":{"type":"illegal_argument_exception","reason":"term: 4\" to_5\" analyzed to a token (4\"_to_5\") with position increment != 1 (got: 0)"}}},"status":400}`

Magento Version : EE 2.3.4

ElasticSuite Version : ~2.8.0, 2.9.x-dev

Environment :  elasticsearch  6.8 

Third party modules : N/A

### Steps to reproduce

1. Create a new Thesaurus, active on all stores
2. Add a thesaurus name "4Inch to 5Inch"
3. Add the synonym terms _**4" to 5",4inch to 5inch**_

### Expected result

1. The Thesaurus index should be rebuilt

### Actual result

1. Thesaurus reindexing fails with the error mentionned above

**Comment by rbayet** [comment]
Flagging this as a bug because the initial need is legitimate.
And if the feature cannot support it, it should be handled more smoothly ...

But I'm not sure that there is a valid approach_system wide_ (indexing, searching) to be able to index and search for `"` (the default analyzers will certainly consider the character as to be ignored).

**Comment by rbayet** [comment]
OK, here are the corresponding settings for the use case 

```
array (
  'number_of_shards' => 1,
  'number_of_replicas' => 0,
  'requests.cache.enable' => true,
  'analysis' => 
  array (
    'filter' => 
    array (
      'shingle' => 
      array (
        'type' => 'shingle',
        'output_false' => true,
        'token_separator' => '_',
        'max_shingle_size' => 10,
      ),
      'type_filter' => 
      array (
        'type' => 'keep_types',
        'types' => 
        array (
          0 => 'SYNONYM',
        ),
      ),
      'synonym' => 
      array (
        'type' => 'synonym',
        'synonyms' => 
        array (
          0 => '4" to_5",4inch_to_5inch',
        ),
      ),
    ),
    'analyzer' => 
    array (
      'synonym' => 
      array (
        'tokenizer' => 'whitespace',
        'filter' => 
        array (
          0 => 'lowercase',
          1 => 'shingle',
          2 => 'synonym',
          3 => 'type_filter',
        ),
      ),
      'expansion' => 
      array (
        'tokenizer' => 'whitespace',
        'filter' => 
        array (
          0 => 'lowercase',
          1 => 'shingle',
          2 => 'type_filter',
        ),
      ),
    ),
  ),
)
```

This is pretty much [this issue](https://github.com/elastic/elasticsearch/issues/29426) and due to the fact that the `whitespace` tokenizer is applied to `4" to_5"` generating at least 2 tokens while `4inch_to_5inch` only generates one.

The original source of the error is that the synonym filter content is incorrectly built, it _might_ work with `4"_to_5",4inch_to_5inch` (no whitespaces = no extra tokens).
This is a regex issue.

**Comment by rbayet** [comment]
Hello @rohitkundale,

PR #1745 will fix the thesaurus indexing issue.
And searching literally for `4" to 5"` will correctly also search for `4inch to 5inch`.

But once you apply the fix _(either by installing our incoming 2.8.5 release or by applying it with a composer patch)_, if you still have some issues, please consider what happens on the analysis chain for `4" to 5"` (or `4inch to 5inch` for that matter) with [a tool like cerebro](https://github.com/lmenezes/cerebro). 

The `"` double quote character will be : 
- stripped by the `reference` analyser
- stripped by the `standard` analyser 
- partially kept, but generating some strange `shingle`s
- stripped by the `whitespace` analyser
- kept by the `sortable` analyser (but it does not have any impact on the fulltext search)

Long story short : by default, that `"` character is **simply not there** for the `standard` and `whitespace` analyser.
Your solution to use the thesaurus will work partially, but you might end up having to add a lot of rules.

Regards,
